src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.cs

```diff
@@ -130,5 +148,64 @@ private void OnErrorListPropertyChanged(object sender, PropertyChangedEventArgs
                 AddTableSourceIfNecessary(this.Workspace.CurrentSolution);
             }
         }
+
+        private void OnErrorListFullSolutionAnalysisToggled(object sender, AnalysisToggleStateChangedEventArgs e)
+        {
+            var newOptions = _optionService.GetOptions()
+                .WithChangedOption(RuntimeOptions.FullSolutionAnalysis, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.CSharp, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.VisualBasic, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, TypeScriptLanguageName, e.NewState);
+            
+            _optionService.SetOptions(newOptions);
+        }
+
+        private void OnOptionChanged(object sender, OptionChangedEventArgs e)
+        {
+            Contract.ThrowIfFalse(_errorList is IErrorList2);
+
+            if (e.Option == RuntimeOptions.FullSolutionAnalysis || e.Option == ServiceFeatureOnOffOptions.ClosedFileDiagnostic)
+            {
+                var analysisDisabled = !(bool)e.Value;
+                if (analysisDisabled)
+                {
+                    ((IErrorList2)_errorList).AnalysisToggleState = false;
+                }
+            }
+        }
+
+        internal static bool ErrorListHasFullSolutionAnalysisButton()
+        {
+            try
+            {
+                // Full solution analysis option has been moved to the error list from Dev14 Update3.
+                // Use reflection to check if the new interface "IErrorList2" exists in Microsoft.VisualStudio.Shell.XX.0.dll.
+                return Assembly.GetAssembly(typeof(ErrorHandler)).GetType("Microsoft.Internal.VisualStudio.Shell.IErrorList2") != null;
```

Comment:
IErrorList2 is accessed in the constructor only if this method returns true, i.e. type exists. I tested it after removing the below temporary IErrorList2 interface and using the new private MS.VS.Shell.14.0 nupkg which has this interface. F5 in Roslyn dev hive shows the old 'Enable full solution analysis' options in Tools Options, and installing the Roslyn VSIX on a machine with the newer MS.VS.Shell.14.0 in GAC causes the new functionality to light up and options are hidden in Tools Options.